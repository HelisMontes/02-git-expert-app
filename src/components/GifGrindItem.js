import React from 'react'

export const GifGrindItem = ({ title, url }) => {
    return (
        <div className='card'>
            <img src={url} alt='Title'/>
            <p>{title}</p>
        </div>
    )
}
