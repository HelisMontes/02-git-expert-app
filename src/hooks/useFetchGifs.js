import { useEffect, useState } from "react"
import { searchApi } from "../helpers/searchApi"

export const useFetchGifs = ( categories ) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })
    useEffect(() => {
        searchApi( categories )
            .then( imgs => {
                setState({
                    data: imgs,
                    loading: false
                });
            });
    }, [ categories ]) 

    return state; // :{ data:[], loading: true } 
}