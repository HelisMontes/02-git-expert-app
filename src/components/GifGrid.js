import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGrindItem } from './GifGrindItem'
import PropTypes from 'prop-types';

export const GifGrid = ({ categories }) => {
    const { loading, data } = useFetchGifs( categories );
    return (
        <>
            <h3 className='animate__animated animate__flipInX'>{categories}</h3>
            {loading && <p className='animate__animated animate__flash'>Loading</p>}
            <div className='card-grind'>
                {
                    data.map( img =>(
                        <GifGrindItem 
                            key = {img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.protoType = {
    categories: PropTypes.string.isRequired
}