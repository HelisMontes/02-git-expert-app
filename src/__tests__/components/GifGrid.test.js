import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Prueba en el componente de GifGrid', () => {
    const categories = 'Son Goku';
    test('Cargar el componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid categories={ categories }/>)
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de mostrar un item cuando se cargan las imagnes de useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://media0.giphy.com/media/eo8UrZwpvjPHi/giphy.gif?cid=2c52d32dnl4k0lwrknihk88lwxo4qz86ygfv8daxk0em8ndn&rid=giphy.gif',
            title: 'rurouni kenshin GIF'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid categories={ categories }/>)
        expect(wrapper).toMatchSnapshot();
    })
})
