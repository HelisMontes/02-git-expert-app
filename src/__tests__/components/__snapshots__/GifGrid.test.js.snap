// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente de GifGrid Cargar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInX"
  >
    Son Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grind"
  />
</Fragment>
`;

exports[`Prueba en el componente de GifGrid Debe de mostrar un item cuando se cargan las imagnes de useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInX"
  >
    Son Goku
  </h3>
  <div
    className="card-grind"
  >
    <GifGrindItem
      id="ABC"
      key="ABC"
      title="rurouni kenshin GIF"
      url="https://media0.giphy.com/media/eo8UrZwpvjPHi/giphy.gif?cid=2c52d32dnl4k0lwrknihk88lwxo4qz86ygfv8daxk0em8ndn&rid=giphy.gif"
    />
  </div>
</Fragment>
`;
